pkgs/development/tools/database/sqldeveloper/default.nix

```diff
@@ -0,0 +1,48 @@
+{ stdenv, makeWrapper, requireFile, unzip, oraclejdk7}:
+
+stdenv.mkDerivation rec {
+  version = "4.0.3.16.84";
+  name = "sqldeveloper-${version}";
+
+  src = requireFile {
+    name = "${name}-no-jre.zip";
+    url = http://www.oracle.com/technetwork/developer-tools/sql-developer/downloads/index.html;
+    sha256 = "1qbqjkfda7xry716da2hdbbazks96rgyslrw1lw0azmqdp1mir7g";
+  };
+
+  buildInputs = [ makeWrapper unzip ];
+
+  buildCommand = ''
+    mkdir -p $out/bin
+    # patch to be able to install a sqldeveloper wrapper script compliant with nix's bin folder once installed
+    echo -e '#!/bin/bash\ncd "`dirname $0`"/../sqldeveloper/bin && bash sqldeveloper $*' >> $out/bin/sqldeveloper
```

Comment:
It's not a check, it's to make it ok between nix (to be able to call sqldeveloper without it's absolute path) and the slightly modified `sqldeveloper.sh` wrapper (which do not permit extensibility with the wrapProgram).

From `$out/bin`, the `dirname $0"/../sqldeveloper/bin` folder is the actual sqldeveloper binary.

I do not want to differ too much from the current nixos approach.
But I do in generating this bash script.
But because the existing one cannot be inside `$out/bin` or else it won't work.

I'm not happy to generate it myself the `sqldeveloper.sh` so I try to make it as much as possible the same as the original one (to ease the maintenance later).